### ISSUE TITLE
translate-cloak doesn't work in transclude

### ISSUE THREAD

**Issue by shaharkazaz** [angular-translate/angular-translate#1556]
### Translate-Cloak doesn't work in transclude

When transcluding an element with translation on it & translate-clock, the translation is being converted to text but the "Translate-Cloak" class isn't being removed.
### Your environment
- angular-translate v2.11.1
- angular v1.5.7
- chrome v51.0.2704.103 (64-bit)
### Steps to reproduce

Issue reproduction in [plunker](http://plnkr.co/edit/EVuvp2pp9wzVTyu8YYCW)
### Expected behaviour

The "Translate-Cloak" class should be removed if translation is found and the promise is resolved.
### Actual behaviour

The class isn't being removed.
### Possible lead

I tried to debug the thing a little bit and what I saw is that the tElement  and the iElement are not the same one, the remove class should be called from the link function, because of the transclude behaviour.

**Comment by mtwar** [comment]
I have a fix for this and am trying to figure out how to pull request on this repository. Here is my local commit message and code fix:

fix(translateCloak): incorrect element reference

In some instances, the element actually injected and populated in the DOM appears to be a copy of the template element passed to $compile. In this situation, although translate-cloak's logic operates normally, the changes apply to the template, not the actual DOM content, so the cloak is never removed. This fix changes the operative element reference post-compile to the element reference provided to the postLink function, which always references the actual DOM content, thereby propagating the changes as expected. The erroneous behavior was observed when applying translate-cloak inside a directive template.

`function translateCloakDirective($translate, $rootScope) {

  'use strict';

  return {
    compile: function (tElement) {
      var applyCloak = function (element) {
        element.addClass($translate.cloakClassName());
      },
      removeCloak = function (element) {
        element.removeClass($translate.cloakClassName());
      };
      applyCloak(tElement);

      return function linkFn(scope, iElement, iAttr) {
		//Create bound functions that incorporate the active DOM element.
		var iRemoveCloak = removeCloak.bind(this, iElement), iApplyCloak = applyCloak.bind(this, iElement);
		$translate.onReady(iRemoveCloak);
        if (iAttr.translateCloak && iAttr.translateCloak.length) {
		  
          // Register a watcher for the defined translation allowing a fine tuned cloak
          iAttr.$observe('translateCloak', function (translationId) {
            $translate(translationId).then(iRemoveCloak, iApplyCloak);
          });
          // Register for change events as this is being another indicicator revalidating the cloak)
          $rootScope.$on('$translateChangeSuccess', function () {
            $translate(iAttr.translateCloak).then(iRemoveCloak, iApplyCloak);
          });
        }
      };
    }
  };
}`

**Comment by knalli** [comment]
@mtwar  Thanks. Please have a look at https://github.com/angular-translate/angular-translate/blob/master/CONTRIBUTING.md

**Comment by IckleChris** [comment]
@mtwar are you looking to PR this fix soon? Appears to resolve the issue for me.

**Comment by mtwar** [comment]
@knalli I have looked at the guide, but I'm currently running into 403 and permission denied when pushing my branch. I do have some time to deal with this now that our product at work is in GA though :) 

I also found another issue with the logic here and can include that as well; basically, the $translate.onReady handler needs to be assigned in an else branch off the iAttr.translateCloak conditions, otherwise it is always called and the cloak removed onReady, even if the specified translationId is not available yet. That change, along with a change I am submitting for $translate.refresh to reassign the language promise, will result in translate-cloak working appropriately with the $translatePartialLoader by gradually uncloaking the page as controls load their strings files.
